src/query_plan/operation.rs

```diff
@@ -1125,6 +1275,27 @@ pub(crate) mod normalized_inline_fragment_selection {
         pub(crate) selection_set: NormalizedSelectionSet,
     }
 
+    impl NormalizedInlineFragmentSelection {
+        pub(crate) fn with_updated_selection_set(
+            &self,
+            selection_set: Option<NormalizedSelectionSet>,
+        ) -> Self {
+            Self {
+                inline_fragment: self.inline_fragment.clone(),
+                //FIXME
```

Comment:
What needs to be FIXME'd here?

Reply:
I guess the use of `unwrap`? Not sure what the best fix would be though. In the one (eventual) call site, `None` is only passed here if `NormalizedSelection::selection_set` returned `None` for the same selection.